src/s4u/s4u_mutex.cpp

```diff
@@ -14,17 +14,18 @@
 using namespace simgrid;
 
 s4u::Mutex::Mutex() {
-    _mutex = simcall_mutex_init();  
+    smx_mutex_t smx_mutex = simcall_mutex_init();  
+    _mutex = std::shared_ptr<s_smx_mutex>(smx_mutex, [](smx_mutex_t mutex){ SIMIX_mutex_destroy(mutex); });
```

Comment:
In fact, we can do this:

    _mutex = std::shared_ptr<s_smx_mutex>(smx_mutex, SIMIX_mutex_destroy);

which is probably simplet :)